src/python/pants/backend/jvm/tasks/coverage/manager.py

```diff
@@ -89,12 +96,29 @@ class InvalidCoverageEngine(Exception):
 
   def get_coverage_engine(self, task, output_dir, all_targets, execute_java):
     options = task.get_options()
-    if options.coverage or options.coverage_processor or options.is_flagged('coverage_open'):
+    enable_scoverage = ScoveragePlatform.global_instance().get_options().enable_scoverage
+    processor = options.coverage_processor
+
+    if (processor == 'scoverage' and not enable_scoverage):
+      raise self.InvalidCoverageEngine("Cannot set processor to scoverage without first enabling "
+                                       "scoverage (by passing --scoverage-enable-scoverage option)")
+
+    if enable_scoverage:
+      if processor is None:
+        logger.info("Scoverage is enabled. Setting coverage engine to scoverage.")
+      elif processor != 'scoverage':
+        logger.warning(f"Scoverage is enabled. Cannot use {processor} as the engine. Setting "
```

Comment:
IMO, this should just be an error asking someone to set the coverage processor to match. Having multiple flags is confusing, and allowing them to disagree is even more confusing.